testdata/gen_deb.py

```diff
@@ -30,30 +30,45 @@ def AddArFileEntry(fileobj, filename, content=''):
   if len(content) % 2 != 0:
     fileobj.write('\n')  # 2-byte alignment padding
 
+
+def add_file_to_tar(tar, filename, content):
+   tarinfo = tarfile.TarInfo(filename)
+   tarinfo.size = len(content)
+   tar.addfile(tarinfo, fileobj=StringIO(content))
+
+
+def get_metadata(pkg_name, content=[]):
+  if content:
+     return '\n'.join(content)
+  else:
+     return '\n'.join([
+         'Package: {0}'.format(pkg_name),
+         'Description: Just a dummy description for dummy package {0}'.format(pkg_name),
+     ])
+
+
 if __name__ == '__main__':
+  pkg_name = sys.argv[1]
+
   # Create data.tar
   tar = StringIO()
   with tarfile.open('data.tar', mode='w', fileobj=tar) as f:
     tarinfo = tarfile.TarInfo('usr/')
     tarinfo.type = tarfile.DIRTYPE
     f.addfile(tarinfo)
-    tarinfo = tarfile.TarInfo('usr/titi')
-    f.addfile(tarinfo, fileobj=StringIO('toto\n'))
+    add_file_to_tar(f, 'usr/{0}'.format(pkg_name), 'toto\n')
   data = tar.getvalue()
   tar.close()
   # Create control.tar
   tar = StringIO()
   with tarfile.open('control.tar', mode='w', fileobj=tar) as f:
-    tarinfo = tarfile.TarInfo('control')
-    f.addfile(tarinfo, fileobj=StringIO('\n'.join([
-        'Package: test'
-        'Description: Just a dummy test'
-        ])))
+    metadata = get_metadata(pkg_name=pkg_name, content=sys.argv[2:-1])
+    add_file_to_tar(f, 'control', metadata)
   control = tar.getvalue()
   tar.close()
 
   # Write the final AR archive (the deb package)
-  with open(sys.argv[1], 'w') as f:
+  with open(sys.argv[-1], 'w') as f:
```

Comment:
I know its just a test util, but maybe use a flag parsing library or set a global instead of addressing the arg by position.